docs/getting-started-guides/locally.md

```diff
@@ -129,6 +132,19 @@ cluster/kubectl.sh create -f docs/user-guide/pod.yaml
 
 Congratulations!
 
+### Hacking hack-local-up cluster to test more sophisticated deployments, such as HA.
+
+To test highly available apiservers, you can modify local-up-cluster.sh as follows.
+
+1) Write a kubernetes token file which has a kubelet user with a token defined.  Lets call it /tmp/mytokens for concreteness.
+
+2) Launch the apiserver with the `--token-auth-file=/tmp/mytokens` and `--apiserver-node-port=30279` options.  Of course 30279 is just an example port.
```

Comment:
I don't understand what nodeports prove here.  At an implementation level
it is pretty much identical whether you hit node-ip:node-port or
service-ip:service-port.  What are you trying to prevent?

Forget DNS - the service IP will redirect anyone who uses it to the
apiserver(s).  The only things that can't use this are the kube-proxy
itself and kubelet.

On Wed, Sep 23, 2015 at 8:24 AM, jay vyas <notifications@github.com> wrote:

> In docs/getting-started-guides/locally.md
> <https://github.com/kubernetes/kubernetes/pull/13978#discussion_r40216900>
> :
>
> > @@ -129,6 +132,19 @@ cluster/kubectl.sh create -f docs/user-guide/pod.yaml
> >
> >  Congratulations!
> >
> > +### Hacking hack-local-up cluster to test more sophisticated deployments, such as HA.
> > +
> > +To test highly available apiservers, you can modify local-up-cluster.sh as follows.
> > +
> > +1) Write a kubernetes token file which has a kubelet user with a token defined.  Lets call it /tmp/mytokens for concreteness.
> > +
> > +2) Launch the apiserver with the `--token-auth-file=/tmp/mytokens` and `--apiserver-node-port=30279` options.  Of course 30279 is just an example port.
>
> (1) lavalamp's question: NodePort is necessary because we want to test
> that the kubelet is properly connecting through the exposed address,
> (2) thockin's question: in this example, its token validation that is used
> for connecting to the service endpoint.
>
> I guess your point around nodeport~kubernetes.local is that, w/ cluster
> DNS, nodeport isn't required at all?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13978/files#r40216900>.
>